frontend/simple/main.js

```diff
@@ -17,7 +19,42 @@ Vue.use(Router)
 Vue.use(VeeValidate)
 
 Vue.events = new Vue() // global event bus, use: https://vuejs.org/v2/api/#Instance-Methods-Events
-
+/*
+  The following are reusable guard for routes
+  the 'guard' defines how the route is blocked and the redirect determines the redirect behavior
+  when a route is blocked
+ */
+// Check if user is logged in
+var loginGuard = {
+  guard (store) {
+    return !store.state.loggedIn
+  },
+  redirect (to, from) {
+    return { path: '/signup', query: { next: to.path } }
+  }
+}
+// Check if user has a group to invite users to
+var inviteGuard = {
+  guard (store) {
+    return !store.state.currentGroupId
+  },
+  redirect (to, from) {
+    return { path: '/new-group' }
+  }
+}
+function createEnterGuards (store, ...guards) {
+  return function (to, from, next) {
+    for (let i = 0; i < guards.length; i++) {
+      let current = guards[i]
+      if (current.guard(store)) {
+        return next(current.redirect(to, from))
+      } else {
+        continue
+      }
+    }
+    return next()
```

Comment:
There are several ways this function can be simplified:

- Use `for of` style to look through array's values instead using an index
- No need to use `return` at the end, can just call `next()`
- No need for `else { continue }`